### PR TITLE
Add async function support

### PR DESCRIPTION
**Pull request by moeen**
Add async function detection. #22

### CODE REVIEW

**Comment by olstenlarck** [comment]
also, we were thinking about exposing options object as second argument, or something like that, because it's edge case and most may want just want is it genfn or normal (no mmatter async or not)

right, @jonschlinkert?

**Comment by jonschlinkert** [comment]
> also, we were thinking about exposing options object as second argument,

I haven't decided on the best way to handle this. We can't merge this PR though, as it would break a lot of code. I'm leaning towards exposing another method for this.

**Comment by def14nt** [comment]
Maybe we could add "hooks" support? That would not break existing code and would work great.
Think about this:
```js
kindOf(2);
kindOf.register(function a(b) { if(isAsyncFunction(b)) return `AsyncFunction`; return false; });
```
Sort of thing, if a hook returns false, the check does not stop.

**Comment by def14nt** [comment]
I'll submit a PR with the hooks support.

**Comment by def14nt** [comment]
Boom. See what you think!

File: index.js [hunk 1]
```diff
@@ -10,7 +10,13 @@ module.exports = function kindOf(val) {
   if (type === 'number') return 'number';
   if (type === 'symbol') return 'symbol';
   if (type === 'function') {
-    return isGeneratorFn(val) ? 'generatorfunction' : 'function';
+    if (isGeneratorFn(val)) {
+      return 'generatorfunction'
+    } else if (isAsyncFn(val)) {
```
Comment (jonschlinkert) [inline index.js]: no need for `else if` here since the above condition already returns if `true`. Same with the next condition.

**Comment by tunnckoCore** [comment]
Agree.

**Pull request closed without merging**